jerry-main/main-unix-snapshot.c

```diff
@@ -73,6 +93,237 @@ read_file (uint8_t *input_pos_p, /**< next position in the input buffer */
+        {
+          break;
+        }
+
+        if (!strcmp ("global", mode_str_p))
+        {
+          is_snapshot_mode_for_global = true;
+        }
+        else if (!strcmp ("eval", mode_str_p))
+        {
+          is_snapshot_mode_for_global = false;
+        }
+        else
+        {
+          jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Incorrect argument for context mode: %s\n", mode_str_p);
+          return JERRY_STANDALONE_EXIT_CODE_FAIL;
+        }
+        break;
+      }
+      case CLI_OPT_DEFAULT:
+      {
+        const char *file_name_p = cli_consume_string (cli_state_p);
+
+        if (cli_state_p->error == NULL)
+        {
+          source_length = read_file (source_p, file_name_p);
+
+          if (source_length == 0)
+          {
+            jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Input file is empty\n");
+            return JERRY_STANDALONE_EXIT_CODE_FAIL;
+          }
+
+          number_of_files++;
+        }
+        break;
+      }
+      default:
+      {
+        cli_state_p->error = "Internal error";
+        break;
+      }
+    }
+  }
+
+  if (cli_state_p->error != NULL)
+  {
+    if (cli_state_p->arg != NULL)
+    {
+      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s %s\n", cli_state_p->error, cli_state_p->arg);
+    }
+    else
+    {
+      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", cli_state_p->error);
+    }
+
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+  if (number_of_files != 1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: No input file specified!\n");
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+  jerry_init (flags);
+
+  if (!jerry_is_valid_utf8_string (source_p, (jerry_size_t) source_length))
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: Input must be a valid UTF-8 string.\n");
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+
+  size_t snapshot_size = jerry_parse_and_save_snapshot ((jerry_char_t *) source_p,
+                                                        source_length,
+                                                        is_snapshot_mode_for_global,
+                                                        false,
+                                                        output_buffer,
+                                                        sizeof(output_buffer) / sizeof(uint32_t));
+  if (snapshot_size == 0)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: Generating snapshot failed!\n");
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+  FILE *snapshot_file_p = fopen (output_file_name_p, "w");
+  if (snapshot_file_p == NULL)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: Unable to write snapshot file: '%s'\n", output_file_name_p);
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+  fwrite (output_buffer, sizeof (uint8_t), snapshot_size, snapshot_file_p);
+  fclose (snapshot_file_p);
+
+  printf ("Created snapshot file: '%s' (%d bytes)\n", output_file_name_p, snapshot_size);
```

Comment:
cppcheck signals signed/unsigned int mismatch here ("warning(invalidPrintfArgType_sint): %d in format string (no. 2) requires a signed integer given in the argument list.")